openedx/core/djangoapps/user_authn/views/login.py

```diff
@@ -272,6 +275,18 @@ def _track_user_login(user, request):
     )
 
 
+def _handle_edx_org_user_google_auth_flow(user):
+    """
+    If `user` email belongs to `@edx.org` and that email does not exist in `AllowedAuthEdxUser`
+    then `user` must login through `edx.org` Google account
+    """
+    if user and waffle().is_enabled(ENABLE_LOGIN_USING_THIRDPARTY_AUTH_ONLY):
+        user_belongs_to_edx_org = user.email.split('@')[1].strip().lower() == AllowedAuthEdxUser.EDX_EMAIL_DOMAIN
+        user_db_qs = AllowedAuthEdxUser.objects.filter(email=user.email)
+        if user_belongs_to_edx_org and not user_db_qs.exists():
+            raise AuthFailedError(_('As an edx.org user, You must login with your edx.org Google account.'))
```

Comment:
edx-platform is an open source product used by many organizations around the world. We should not be hardcoding behavior for edx.org.

Reply:
To clarify: this PR would be fine if it were changed to make the particular domains that are forced to log in via SSO configurable, so that edx.org could be configured to force `@edx.org` emails through SSO.

Reply:
Thanks @ormsbee and @cpennington, Totally agreed regarding your suggestions. We are adding new models in https://github.com/edx/edx-platform/pull/22292 to make this whole approach generic. Once https://github.com/edx/edx-platform/pull/22292 is done, this code will be updated accordingly.